File: 02-primes/primes.jl
function isprime(primes, x)
    for d in primes
        if d * d > x return true end
        if x % d == 0 return false end
    end
end

function main()
    primes = [2, 3, 5, 7]
    n = parse(Int64, ENV["MAXN"])
    p = 9
    while true
        if isprime(primes, p)
            append!(primes, p)
            if size(primes, 1) == n
                println("primes[$n] = $(primes[n])")
                break
            end
        end
        p += 2
    end
end

main()

File: 02-primes/primes.pl
$target = $ENV{'MAXN'};

@primes = (2, 3, 4, 5);

sub is_prime {
    my ($x) = @_;
    foreach $d (@primes) {
        return 1 if ($d * $d > $x);
        return 0 if ($x % $d == 0);
    }
}

for ($n = 9; 1; $n += 2) {
    if (is_prime($n)) {
        push @primes, $n;
        if (@primes == $target) {
            print "primes[$target] = " . $primes[$target-1] . "\n";
            last;
        }
    }
}
